test/gurka/test_indoor.cc

```diff
@@ -66,37 +70,49 @@ gurka::map Indoor::map = {};
 std::string Indoor::ascii_map = {};
 gurka::nodelayout Indoor::layout = {};
 
-Api api;
-rapidjson::Document d;
-
 /*************************************************************/
 
 TEST_F(Indoor, NodeInfo) {
   baldr::GraphReader graphreader(map.config.get_child("mjolnir"));
 
   auto node_id = gurka::findNode(graphreader, layout, "E");
-  EXPECT_EQ(graphreader.nodeinfo(node_id)->type(), baldr::NodeType::kBuildingEntrance);
+  const auto* node = graphreader.nodeinfo(node_id);
+  EXPECT_EQ(node->type(), baldr::NodeType::kBuildingEntrance);
+  EXPECT_TRUE(node->access() & baldr::kPedestrianAccess);
 
   node_id = gurka::findNode(graphreader, layout, "I");
+  node = graphreader.nodeinfo(node_id);
   EXPECT_EQ(graphreader.nodeinfo(node_id)->type(), baldr::NodeType::kElevator);
+  EXPECT_TRUE(node->access() & baldr::kPedestrianAccess);
 }
 
 TEST_F(Indoor, DirectedEdge) {
   baldr::GraphReader graphreader(map.config.get_child("mjolnir"));
 
-  auto directededge = std::get<1>(gurka::findEdgeByNodes(graphreader, layout, "B", "C"));
+  const auto* directededge = std::get<1>(gurka::findEdgeByNodes(graphreader, layout, "B", "C"));
   EXPECT_EQ(directededge->use(), baldr::Use::kSteps);
+  EXPECT_TRUE(directededge->forwardaccess() & baldr::kPedestrianAccess);
+  EXPECT_TRUE(directededge->reverseaccess() & baldr::kPedestrianAccess);
 
   directededge = std::get<1>(gurka::findEdgeByNodes(graphreader, layout, "G", "H"));
   EXPECT_EQ(directededge->use(), baldr::Use::kElevator);
+  EXPECT_TRUE(directededge->forwardaccess() & baldr::kPedestrianAccess);
+  EXPECT_TRUE(directededge->reverseaccess() & baldr::kPedestrianAccess);
 
   directededge = std::get<1>(gurka::findEdgeByNodes(graphreader, layout, "K", "L"));
   EXPECT_EQ(directededge->use(), baldr::Use::kEscalator);
+  EXPECT_TRUE(directededge->forwardaccess() & baldr::kPedestrianAccess);
+  EXPECT_TRUE(directededge->reverseaccess() & baldr::kPedestrianAccess);
 
   directededge = std::get<1>(gurka::findEdgeByNodes(graphreader, layout, "D", "E"));
   EXPECT_EQ(directededge->indoor(), false);
+  EXPECT_TRUE(directededge->forwardaccess() & baldr::kPedestrianAccess);
+  EXPECT_TRUE(directededge->reverseaccess() & baldr::kPedestrianAccess);
+
   directededge = std::get<1>(gurka::findEdgeByNodes(graphreader, layout, "E", "F"));
   EXPECT_EQ(directededge->indoor(), true);
+  EXPECT_TRUE(directededge->forwardaccess() & baldr::kPedestrianAccess);
+  EXPECT_TRUE(directededge->reverseaccess() & baldr::kPedestrianAccess);
```

Comment:
i added some hardening for all of these. what i did first was add teh routing test below but it failed immediately to even route at all. this made me immediately think that the access was messed up, so to check that i hardened all of these tests to see if pedestrians were even allowed on these edges/nodes. it turned out that no they weren't in some cases which was what the lua change fixed